### PR TITLE
Fix failing uploads to lookaside cache

### LINKED ISSUE #801: Lookaside cache upload fails with HTTP code "400 Bad Request"

### PR DESCRIPTION
**Pull request by nforro**
I'm not really sure about this. The requests still fail randomly with `Remote end closed connection without response`.
I'm still considering implementing the upload with `pycurl` after all, because it seems to work without issues.

Fixes #801.

### CODE REVIEW

**Comment by nforro** [comment]
I think I solved that issue by enabling [opportunistic authentication](https://pypi.org/project/requests-gssapi/#opportunistic-authentication). So the only downside now is the infinite progressbar, which looks a bit weird (especially because it doesn't show transferred bytes).

**Comment by nforro** [comment]
Excerpt from the output:
```
INFO: Uploading vim-go-1.24.tar.gz to lookaside cache
    [     <=>                      ]     in 00:00:01 
INFO: Running checks on source files using 'licensecheck'
```

File: rebasehelper/helpers/lookaside_cache_helper.py [hunk 1]
```diff
@@ -167,24 +167,33 @@ def post(check_only=False):
...
         if state.strip() == b'Available':
             # already uploaded
+            logger.info('Source is already present in lookaside cache')
             return
 
         logger.info('Uploading %s to lookaside cache', path)
         try:
-            post()
-        except requests.exceptions.ConnectionError as e:
+            try:
```
Comment (FrNecas) [inline rebasehelper/helpers/lookaside_cache_helper.py:216]: Is this inner try-finally block necessary? Shouldn't it be sufficient to keep the finally part connected to the outer `try-except-finally` block?

**Review by FrNecas — APPROVED** [review]
If the problem with `Remote end closed connection without response` was solved by opportunistic authentication, it looks good to me and we can merge :)

**Pull request merged**